src/__integrations__/creditCard.devServer.integration.test.ts

```diff
@@ -0,0 +1,149 @@
+    await page.setViewport({ width: 1200, height: 907 });
+
+    const creditCardFieldS = '#creditcardnumber';
+
+    await page.focus(creditCardFieldS);
+    await page.keyboard.type('5555555555554444');
+    const checked = await page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'));
+    expect(checked).toMatch(/1/);
+  });
+
+  it('should handle Visa cards', async () => {
+    const page = await myBrowser.newPage();
+
+    await page.goto(`http://${HOST}:${PORT}/index.html?p=1234`);
+    await page.setViewport({ width: 1200, height: 907 });
+
+    const creditCardFieldS = '#creditcardnumber';
+
+    await page.focus(creditCardFieldS);
+    await page.keyboard.type('4111111111111111');
+    const checked = await page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'));
+    expect(checked).toMatch(/1/);
+    await page.close();
+  });
+
+  it('should handle Maestro cards', async () => {
+    // Start the browser environment
+    const page = await myBrowser.newPage();
+
+    await page.goto(`http://${HOST}:${PORT}/index.html?p=1234`);
+    await page.setViewport({ width: 1200, height: 907 });
+
+    const creditCardFieldS = '#creditcardnumber';
+    const creditCardInput = await page.$(creditCardFieldS);
+
+    await page.focus(creditCardFieldS);
+    await page.keyboard.type('6759649826438453');
+    await expect(page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'))).resolves.toMatch(/1/);
+
+    // Delete inserted PAN
+    await creditCardInput?.click({ clickCount: 3 });
+    await creditCardInput?.press('Backspace');
+
+    // Different length
+    await page.keyboard.type('6799990100000000019');
+    await expect(page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'))).resolves.toMatch(/1/);
+    await page.close();
+  });
+
+  it('should handle American Express cards', async () => {
+    const page = await myBrowser.newPage();
+
+    await page.goto(`http://${HOST}:${PORT}/index.html?p=1234`);
+    await page.setViewport({ width: 1200, height: 907 });
+
+    const creditCardFieldS = '#creditcardnumber';
+
+    await page.focus(creditCardFieldS);
+    await page.keyboard.type('340024388482878');
+    await expect(page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'))).resolves.toMatch(/1/);
+    await page.close();
+  });
+
+  it('should remove data-checked attribute when the card number is removed', async () => {
+    const page = await myBrowser.newPage();
+
+    await page.goto(`http://${HOST}:${PORT}/index.html?p=1234`);
+    await page.setViewport({ width: 1200, height: 907 });
+
+    const creditCardFieldS = '#creditcardnumber';
+    const creditCardInput = await page.$(creditCardFieldS);
+
+    await page.focus(creditCardFieldS);
+    await page.keyboard.type('5555555555554444');
+    const checked = await page.$eval(creditCardFieldS, element => element.getAttribute('data-checked'));
+    expect(checked).toMatch(/1/);
```

Comment:
why in the previously test do you use `toMatch` and here `resolves`  to check `data-checked` equal to `1` ?

Reply:
@pasqualespica  Thank you for pointing out. `page.$eval()` returns a `Promise`, so if I call it _inside_ `expect`, as done in previous test, I have to use `resolves` to guarantee that it resolves. On the other hand, If I `await` _outside_ expect, as done in this test, it's no longer a `Promise` but a resolved value, so using `toMach()` is sufficient. I just experimented with different ways of asserting on `Promises` but the two tests, as you pointed out, are perfectly equivalent